scss/mixins/_forms.scss

```diff
@@ -130,7 +130,11 @@
   .input-group .form-control,
   .input-group .form-select {
     @include form-validation-state-selector($state) {
-      z-index: 3;
+      @if $state == "valid" {
```

Comment:
I don't get why the `z-index` for valid items has to be set as well.
Also, a `z-index` of 3 seems too high to me. If 3 is used for an invalid item we would have to use 4 for a focused element ([demo](https://codepen.io/korki43/pen/eYgjVob?editors=1100)). 

I think something like this should work fine ([demo](https://codepen.io/korki43/pen/YzNjaXG?editors=1000)):

```scss
@include form-validation-state-selector($state) {
  @if $state == "invalid" {
    z-index: 2;
  }
  &:focus {
    z-index: 3
  }
}

```

Ref: [`z-index` scales approach from the docs](https://getbootstrap.com/docs/5.0/extend/approach/#z-index-scales)

Reply:
The initial issue will remain for valid state, that's why. 

To sum up my thought:
- default state is the less important,
- then valid state should be visible ans more important than default, in cases like the initial issue,
- invalid state then takes precendence, since it needs to be as visible as possible,
- then focus should always be as visible as possible.

So maybe start from 1 to 3, but IMHO we need to declare 3 différent values.

Might be donne more elegantly than my suggestion, but that's the target I guess.

Reply:
Ok, I thought `valid` should equal the initial state, but your idea seems more reasonable. In this case I also think 3 values are the way to go.


Something like this should respect everything:
```scss
@include form-validation-state-selector($state) {
  @if $state == "valid" {
    z-index: 1;
  } @else if $state == "invalid" {
    z-index: 2;
  }
  &:focus {
   z-index: 3;
  }
}
```